src/commands/inputs.rs

```diff
@@ -0,0 +1,540 @@
+                LiteralExpr::Boolean(v) => {
+                    return Some((namespace.push(name), Value::Bool(v.value())));
+                }
+                LiteralExpr::String(v) => {
+                    if let Some(text) = v.text() {
+                        let mut buffer = String::new();
+                        text.unescape_to(&mut buffer);
+                        return Some((namespace.push(name), Value::String(buffer)));
+                    } else {
+                        return Some((namespace.push(name), Value::String(Default::default())));
+                    }
+                }
+                LiteralExpr::Integer(v) => {
+                    return Some((namespace.push(name), Value::from(v.value().unwrap_or(0))));
+                }
+                LiteralExpr::Float(v) => {
+                    return Some((namespace.push(name), Value::from(v.value().unwrap_or(0.0))));
+                }
+                LiteralExpr::Struct(v) => {
+                    let map = v
+                        .items()
+                        .filter_map(|item| {
+                            let (name, value) = item.name_value();
+                            self.expression(ty.clone(), Key::empty(), name.text(), &value)
+                        })
+                        .map(|(k, v)| (k.join().expect("key to join"), v))
+                        .collect::<Map<_, _>>();
+
+                    return Some((namespace.push(name), Value::Object(map)));
+                }
+                LiteralExpr::Array(a) => {
+                    let mut values = vec![];
+
+                    for item in a.elements() {
+                        if let Some((key, value)) =
+                            self.expression(ty.clone(), Key::empty(), name, &item)
+                        {
+                            values.push((key.join().expect("key to join"), value));
+                        }
+                    }
+
+                    return Some((
+                        namespace.push(name),
+                        Value::Array(values.into_iter().map(|(_, v)| v).collect()),
+                    ));
+                }
+                LiteralExpr::Map(m) => {
+                    let mut map = Map::new();
+
+                    for item in m.items() {
+                        let (k, v) = item.key_value();
+                        let key_name: String = match k {
+                            Expr::Literal(LiteralExpr::String(k)) => {
+                                if let Some(text) = k.text() {
+                                    let mut buffer = String::new();
+                                    text.unescape_to(&mut buffer);
+                                    buffer
+                                } else {
+                                    String::new()
+                                }
+                            }
+                            _ => k.text().to_string(),
```

Comment:
This case is the `Map` has a non-String key? Should this instead recurse maybe? Wouldn't we want an `Int` map to render as number values instead of strings?

Reply:
Ideally, but serde_json's map is `Map<String, Value>`. JSON objects require the key to be a string. The strings will need to be parsed to the correct type in the consumer code.